Source/Schema.NET/OneOrMany{T}.cs

```diff
@@ -15,131 +16,130 @@ namespace Schema.NET
         : IReadOnlyCollection<T>, IEnumerable<T>, IValues, IEquatable<OneOrMany<T>>
 #pragma warning restore CA1710 // Identifiers should have correct suffix
     {
-        private readonly List<T> collection;
-        private readonly T item;
+        private readonly T[] collection;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="OneOrMany{T}"/> struct.
         /// </summary>
         /// <param name="item">The single item value.</param>
         public OneOrMany(T item)
         {
-            this.collection = null;
-            this.item = item;
-        }
-
-        /// <summary>
-        /// Initializes a new instance of the <see cref="OneOrMany{T}"/> struct.
-        /// </summary>
-        /// <param name="array">The array of values.</param>
-        public OneOrMany(params T[] array)
-            : this(array is null ? null : new List<T>(array))
-        {
-        }
-
-        /// <summary>
-        /// Initializes a new instance of the <see cref="OneOrMany{T}"/> struct.
-        /// </summary>
-        /// <param name="collection">The collection of values.</param>
-        public OneOrMany(IEnumerable<T> collection)
-            : this(collection is null ? null : new List<T>(collection))
-        {
-        }
-
-        /// <summary>
-        /// Initializes a new instance of the <see cref="OneOrMany{T}"/> struct.
-        /// </summary>
-        /// <param name="list">The list of values.</param>
-        public OneOrMany(List<T> list)
-        {
-            if (list is null)
-            {
-                throw new ArgumentNullException(nameof(list));
-            }
-
-            if (list.Count == 1)
+            if (item is null || (item is string itemAsString && string.IsNullOrWhiteSpace(itemAsString)))
             {
                 this.collection = null;
-                this.item = list[0];
+                this.HasOne = false;
             }
             else
             {
-                this.collection = list;
-                this.item = default;
+                this.collection = new[] { item };
+                this.HasOne = true;
             }
         }
 
         /// <summary>
         /// Initializes a new instance of the <see cref="OneOrMany{T}"/> struct.
         /// </summary>
-        /// <param name="list">The list of values.</param>
-        public OneOrMany(IEnumerable<object> list)
+        /// <param name="array">The array of values.</param>
+        public OneOrMany(params T[] array)

```

Comment:
If we care about performance, perhaps we should add overloads for `Span<T>`?

Reply:
Yeah - I was even thinking about using `Span<T>` internally instead of even the array. Unfortunately that would require the `OneOrMany` struct to be a `ref`.